### DISCUSSION TITLE
trusted platform module signer support

### DISCUSSION THREAD

**Discussion by salrashid123** [golang-jwt/jwt#132] [Show and tell]
not a bug or FR but incase any user here is interested, a jwt signer/extension that uses a trusted platform module

https://github.com/salrashid123/golang-jwt-tpm

(just limited alpha quality; (only tested on linux and on rasp-pi)); feel free to PR or issues though i don't know how to use windows TPM issues)

**Comment by oxisto** [comment]
Interesting idea! I have moved this to the discussion section.

**Comment by salrashid123** [comment]
an application is for this  [google cloud IoT auth](https://github.com/salrashid123/iot_tpm_auth) which just uses a JWT but all the examples describes a plain cert on the device.

fwiw, the more generic form is to use pkcs11 interface like [this](https://github.com/salrashid123/go_pkcs11) but its harder to setup (though that'll allow you to potentially sign using an HSM or youbikey or tpm, etc)
